File: wasm/nodejs/message_signing.js
let kaspa = require('./kaspa/kaspa_wasm');
let {
    PrivateKey,
    PublicKey,
    signMessage,
    verifyMessage,
} = kaspa;

kaspa.initConsolePanicHook();

let message = 'Hello Kaspa!';
let privkey = 'B7E151628AED2A6ABF7158809CF4F3C762E7160F38B4DA56A784D9045190CFEF';
let pubkey = 'DFF1D77F2A671C5F36183726DB2341BE58FEAE1DA2DECED843240F7B502BA659';

function runDemo(message, privateKey, publicKey) {
    let signature = signMessage({message, privateKey});

    console.info(`Message: ${message} => Signature: ${signature}`);

    if (verifyMessage({message, signature, publicKey})) {
        console.info('Signature verified!');
    } else {
        console.info('Signature is invalid!');
    }
}

// Using strings:
runDemo(message, privkey, pubkey);
// Using Objects:
runDemo(message, new PrivateKey(privkey), new PublicKey(pubkey));

File: python/examples/message-signing.py
from kaspa import PrivateKey, PublicKey, sign_message, verify_message

if __name__ == "__main__":
    message = "Hello Kaspa!"
    private_key = PrivateKey("b7e151628aed2a6abf7158809cf4f3c762e7160f38b4da56a784d9045190cfef")
    public_key = PublicKey("dff1d77f2a671c5f36183726db2341be58feae1da2deced843240f7b502ba659")

    signature = sign_message(message, private_key)
    print(f'Signature: {signature}')

    valid_sig = verify_message(message, signature, public_key) 
    print('Valid sig' if valid_sig else 'Invalid sig')
